devices/src/virtio/vsock/mod.rs

```diff
@@ -211,3 +211,179 @@ impl VsockChannel for DummyBackend {
+            self.tx_err = err;
+        }
+        pub fn set_pending_rx(&mut self, prx: bool) {
+            self.pending_rx = prx;
+        }
+    }
+    impl VsockChannel for TestBackend {
+        fn recv_pkt(&mut self, _pkt: &mut VsockPacket) -> Result<()> {
+            match self.rx_err.take() {
+                None => {
+                    self.rx_ok_cnt += 1;
+                    Ok(())
+                }
+                Some(e) => Err(e),
+            }
+        }
+        fn send_pkt(&mut self, _pkt: &VsockPacket) -> Result<()> {
+            match self.tx_err.take() {
+                None => {
+                    self.tx_ok_cnt += 1;
+                    Ok(())
+                }
+                Some(e) => Err(e),
+            }
+        }
+        fn has_pending_rx(&self) -> bool {
+            self.pending_rx
+        }
+    }
+    impl VsockEpollListener for TestBackend {
+        fn get_polled_fd(&self) -> RawFd {
+            self.evfd.as_raw_fd()
+        }
+        fn get_polled_evset(&self) -> epoll::Events {
+            epoll::Events::EPOLLIN
+        }
+        fn notify(&mut self, evset: epoll::Events) {
+            self.evset = Some(evset);
+        }
+    }
+    impl VsockBackend for TestBackend {}
+
+    pub struct TestContext {
+        pub cid: u64,
+        pub mem: GuestMemory,
+        pub mem_size: usize,
+        pub device: Vsock<TestBackend>,
+
+        // This needs to live here, so that sending the handler, at device activation, works.
+        _handler_receiver: mpsc::Receiver<Box<EpollHandler>>,
+    }
+
+    impl TestContext {
+        pub fn new() -> Self {
+            const CID: u64 = 52;
+            const MEM_SIZE: usize = 1024 * 1024 * 128;
+            let (sender, _handler_receiver) = mpsc::channel();
+            Self {
+                cid: CID,
+                mem: GuestMemory::new(&[(GuestAddress(0), MEM_SIZE)]).unwrap(),
+                mem_size: MEM_SIZE,
+                device: Vsock::new(
+                    CID,
+                    EpollConfig::new(0, epoll::create(true).unwrap(), sender),
+                    TestBackend::new(),
+                )
+                .unwrap(),
+                _handler_receiver,
+            }
+        }
+
+        pub fn create_epoll_handler_context(&self) -> EpollHandlerContext {
+            const QSIZE: u16 = 2;
+
+            let guest_rxvq = GuestQ::new(GuestAddress(0x0010_0000), &self.mem, QSIZE as u16);
+            let guest_txvq = GuestQ::new(GuestAddress(0x0020_0000), &self.mem, QSIZE as u16);
+            let guest_evvq = GuestQ::new(GuestAddress(0x0030_0000), &self.mem, QSIZE as u16);
+            let rxvq = guest_rxvq.create_queue();
+            let txvq = guest_txvq.create_queue();
+            let evvq = guest_evvq.create_queue();
+
+            // Set up one available descriptor in the RX queue.
+            guest_rxvq.dtable[0].set(0x0040_0000, 44, VIRTQ_DESC_F_WRITE | VIRTQ_DESC_F_NEXT, 1);
```

Comment:
Why not use `VSOCK_PKT_HDR_SIZE instead of 44. It is easier to understand what "available descriptor means" in this case.

Reply:
Done.